Mlem/Views/Shared/Instance/InstanceUptimeView.swift

```diff
@@ -0,0 +1,248 @@
+        VStack(alignment: .leading, spacing: 0) {
+            section { summary }
+            .padding(.top, 16)
+            .padding(.bottom, 10)
+            section("Recent Checks") {
+                recentChecks
+                    .padding(.horizontal)
+                    .padding(.vertical, 15)
+            }
+            .padding(.top, 20)
+            footnote("Mlem will refresh this automatically every 30 seconds.")
+            .padding(.top, 8)
+            .padding(.leading, 6)
+            .padding(.bottom, 30)
+            section("Response Time") {
+                VStack(alignment: .leading, spacing: 4) {
+                    responseTimeChart
+                        .padding(.horizontal, 20)
+                    footnote("Mean average: \(uptimeData.results.map(\.durationMs).reduce(0, +) / uptimeData.results.count)ms")
+                        .padding(.leading, 20)
+                }
+                .padding(.top, 17)
+                .padding(.bottom, 8)
+            }
+            section("Incidents", spacing: 0) {
+                ForEach(uptimeData.downtimes) { event in
+                    if event.id != uptimeData.downtimes.first?.id {
+                        Divider()
+                    }
+                    IncidentRow(event: event, showingExactTime: showingExactTime)
+                    .padding(.vertical, 10)
+                    .padding(.leading)
+                }
+                .onTapGesture {
+                    withAnimation(.easeInOut(duration: 0.2)) {
+                        showingExactTime.toggle()
+                    }
+                }
+            }
+            .padding(.top, 30)
+            if let url = instance.uptimeFrontendUrl {
+                Text("Uptime data fetched from [lemmy-status.org](\(url))")
+                    .font(.footnote)
+                    .foregroundStyle(.secondary)
+                    .padding(.vertical, 8)
+                    .padding(.leading, 6)
+            }
+            if colorScheme == .light {
+                Divider()
+            }
+        }
+        .padding(.horizontal, 16)
+        .background(Color(uiColor: .systemGroupedBackground))
+    }
+    
+    @ViewBuilder
+    var summary: some View {
+        VStack(alignment: .leading) {
+            if let mostRecentOutage = uptimeData.downtimes.first {
+                if uptimeData.results.filter(\.success).count < 15 {
+                    if mostRecentOutage.endTime == nil {
+                        HStack(spacing: 5) {
+                            (Text("\(instance.name) is ") + Text("offline").foregroundColor(.red))
+                                .font(.title2)
+                                .fontWeight(.semibold)
+                            Image(systemName: Icons.uptimeOffline)
+                                .foregroundStyle(.red)
+                        }
+                        footnote("Outage started \(mostRecentOutage.startTime.getRelativeTime()).")
```

Comment:
*nit*

This block gets duplicated three times--maybe move to a helper function?